### PR TITLE
feat!: Move to dart 2.17 and clean up comment.

### PR DESCRIPTION
**Pull request by spydon**
# Description

This updates the min version of Dart to 2.17 and updates to the newest flame_lint.
All analyze issues are fixed in this PR + removing `@param` and `@return`

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [xj] I have updated/added relevant examples in `examples`.

## Breaking Change

- [x] Yes, this is a breaking change.

## Related Issues

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/forge2d/lib/src/dynamics/fixture.dart [hunk 1]
```diff
@@ -86,53 +84,49 @@ class Fixture {
...
-  /// Test a point for containment in this fixture. This only works for convex shapes.
+  /// Test a point for containment in this fixture. This only works for convex
+  /// shapes.
   ///
-  /// @param p a point in world coordinates.
-  /// @return
-  bool testPoint(final Vector2 p) {
-    return shape.testPoint(body.transform, p);
+  /// [point] should be in world coordinates.
+  bool testPoint(final Vector2 point) {
```
Comment (luanpotter) [inline packages/forge2d/lib/src/dynamics/fixture.dart:91]: why this one particular parameter is final and nothing else?

File: packages/forge2d/lib/src/dynamics/contacts/contact_solver.dart [hunk 2]
```diff
@@ -135,11 +135,9 @@ class ContactSolver {
...
   }
 
-  // djm pooling, and from above
-  // TODO(srdjan): make them private.
-  final Transform xfA = Transform.zero();
-  final Transform xfB = Transform.zero();
-  final WorldManifold worldManifold = WorldManifold();
+  final Transform _xfA = Transform.zero();
+  final Transform _xfB = Transform.zero();
+  final WorldManifold _worldManifold = WorldManifold();
```
Comment (luanpotter) [inline packages/forge2d/lib/src/dynamics/contacts/contact_solver.dart:140]: nice!

**Review by luanpotter — APPROVED** [review]
Excellent!

File: packages/forge2d/lib/src/dynamics/fixture.dart (hunk 1 above)
Comment (spydon) [reply packages/forge2d/lib/src/dynamics/fixture.dart:91]: No idea, nothing that comes from this PR though 😅

**Pull request merged**